### ISSUE TITLE
Allow the omission of the superclass

### ISSUE THREAD

**Issue by rauschma** [justinfagnani/mixwith.js#2]
Currently the syntax is:

```js
class MyClass extends mix(MySuperClass).with(MyMixin1, MyMixin2) {
}
```

How about the following?

```js
class MyClass extends mixins(MyMixin1, MyMixin2, MySuperClass) {
}
```

Details:

* This better reflects the order of the inheritance chain
* The last parameter should be optional (you may want to create classes that only assemble mixins and have no superclass)
* The function could do a simple typecheck by enforcing that the mixins must not have a property `prototype`, while the superclass (the optional last parameter) must have this property.

**Comment by Sghazzawi** [comment]
What about the case where, e.g. 
```
MyMixin2  = (superclass) => class extends MixinX(superclass) {} 
```
or 
```
MyMixin2 = (superclass) => MixinX(MixinY(superclass));
```
would the mixin function not support mixing either of those in to the superclass?

**Comment by psayre23** [comment]
If this syntax is acceptable:

    class MyClass extends mix().with(MyMixin1, MyMixin2) {

...then you could throw this check in for mix constructor:

    if(typeof superclass !== "object" || superclass === null) {
        superclass = Object;
    }

That would default to having Object as a superclass...which everything is, so that's technically true.

**Comment by justinfagnani** [comment]
I like that suggestion @psayre23

**Comment by justinfagnani** [comment]
Anyone want to look at #7 ?

**Comment by mvila** [comment]
Unfortunately, the `extends Object` trick doesn't work when transpiled with Babel 6 (tested with the last version). I have filled a bug report about that: [https://phabricator.babeljs.io/T6964](https://phabricator.babeljs.io/T6964).

**Comment by justinfagnani** [comment]
Ouch, good catch @mvila. I suppose I could work around this by defining an empty class to use as the default rather than `Object`. It'd be better to see that fixed though. Thanks!

**Comment by mvila** [comment]
Well, apparently the "bug" will not be fixed... :( So maybe it is better to use an empty class.

**Comment by justinfagnani** [comment]
Looks like we could also use `null`. Shame they closed that issue so fast.

**Comment by mvila** [comment]
Using `null` is not a good option because we will create a class without prototype at all. So, no default constructor and nothing inherited from Object.prototype (like the very useful obj.hasOwnProperty()). This article shows the differences between extending Object, null or nothing: http://www.2ality.com/2015/02/es6-classes-final.html

**Comment by trusktr** [comment]
> That would default to having Object as a superclass...which everything is, so that's technically true.

`Object` can be replaced at runtime, whereas `class {}` always extends from the internal `[[Object]]`, so possibly it's better to use an empty class anyway.

**Comment by justinfagnani** [comment]
Fixed in 2051b85b5a37f97d34eb323e10a36db25e440d36 and 5b66154b6019a8264b3530cf86189fbbf7b04e77
